packages/core/src/components/templates/PeriodInput/PeriodInput.tsx

```diff
@@ -0,0 +1,128 @@
+/*
+ * Axelor Business Solutions
+ *
+ * Copyright (C) 2024 Axelor (<http://axelor.com>).
+ *
+ * This program is free software: you can redistribute it and/or  modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, {useEffect, useMemo, useState} from 'react';
+import {StyleSheet, View} from 'react-native';
+import {Label, useThemeColor} from '@axelor/aos-mobile-ui';
+import {DateInput} from '../../organisms';
+import {getEndOfDay, getStartOfDay} from '../../../utils';
+import {useTranslator} from '../../../i18n';
+
+interface dateInputConfig {
+  date?: Date;
+  onDateChange: (date: Date) => void;
+  readonly?: boolean;
+  required?: boolean;
+}
+
+interface PeriodInputProps {
+  startDateConfig: dateInputConfig;
+  endDateConfig: dateInputConfig;
+  horizontal?: boolean;
+  style?: any;
+}
+
+const PeriodInput = ({
+  startDateConfig,
+  endDateConfig,
+  horizontal = true,
+  style,
+}: PeriodInputProps) => {
+  const I18n = useTranslator();
+  const Colors = useThemeColor();
+
+  const [fromDate, setFromDate] = useState(startDateConfig.date);
+  const [toDate, setToDate] = useState(endDateConfig.date);
+  const [isPeriodError, setIsPeriodError] = useState(false);
+
+  useEffect(() => {
+    setFromDate(startDateConfig.date);
+  }, [startDateConfig.date]);
+
+  useEffect(() => {
+    setToDate(endDateConfig.date);
+  }, [endDateConfig.date]);
+
+  useEffect(() => {
+    setIsPeriodError(
+      fromDate && toDate && getStartOfDay(fromDate) > getEndOfDay(toDate),
+    );
+  }, [fromDate, toDate]);
+
+  const styles = useMemo(() => {
+    return getStyles(horizontal);
+  }, [horizontal]);
+
+  return (
+    <View style={[styles.container, style]}>
+      {isPeriodError && (
+        <Label
+          message={I18n.t('Base_PeriodError')}
+          iconName="exclamation-triangle-fill"
+          color={Colors.errorColor}
```

Comment:
you can give props "type" to error, it will manage iconName and color on it's own
You also have a props "visible" if you want to simplify render condition